src/yaml.zig

```diff
@@ -181,8 +181,7 @@ pub const Value = union(enum) {
             const raw = tree.getRaw(node.start, node.end);
 
             try_int: {
-                // TODO infer base for int
-                const int = std.fmt.parseInt(i64, raw, 10) catch break :try_int;
+                const int = std.fmt.parseInt(i64, raw, 0) catch break :try_int;
```

Comment:
Looks like a good first step. I am wondering though, and asking really since I don't see many YAML files day-to-day except for machine-generated dylib stubs on macOS: is it possible to have int encoded in different bases without a prefix in YAML?

Reply:
They need a prefix in YAML 1.2 and in this implementation.

From the [`parseInt()` documentation](https://ziglang.org/documentation/master/std/#A;std:fmt.parseInt), it infers the base according to the prefix:

"When base [3rd argument] is zero the string prefix is examined to detect the true base:
- A prefix of “0b” implies base=2,
- A prefix of “0o” implies base=8,
-  A prefix of “0x” implies base=16,
-  Otherwise base=10 is assumed."

According to the spec, YAML parses integers in decimal, hexadecimal, and octal bases. However, it should **not** parse an hex/oct without the corresponding prefix.

One interesting bug/feature is that this implementation allows binary base (0b1101), which is [not supported by YAML 1.2](https://github.com/symfony/symfony/issues/8094).

Reply:
Yeah, I know that `std.fmt.parseInt` uses the prefix to infer the base, hence why I was asking if this matches YAML spec. But it seems it does, so perfect then!